tf/modules/monitoring/signup-throttling-alert.tf

```diff
@@ -0,0 +1,39 @@
+resource "google_logging_metric" "api_signup_throttling_applied" {
+  name        = "${var.cluster_name}-api_signup_throttling_applied"
+  description = "A log based metric that detects when api was throttling sign up attempts"
+
+  # Filter for the secondary pod and look for readiness probe failures
+  filter = "resource.labels.cluster_name=\"${var.cluster_name}\" AND resource.labels.container_name:\"api-\" AND severity=ERROR AND resource.type=\"k8s_container\" AND textPayload:\"WARN_SIGNUP_THROTTLED\""
+
+  metric_descriptor {
+    metric_kind = "DELTA"
+    value_type  = "INT64"
+  }
+}
+
+resource "google_monitoring_alert_policy" "alert_api_signup_throttling_applied" {
+  display_name = "(${var.cluster_name}): Signup was throttled"
+  combiner     = "OR"
+  notification_channels = [
+    "${var.monitoring_email_group_name}"
+  ]
+  documentation {
+    content = "Alert triggers when a user tried to create a wikibase.cloud account, but the configured limits were already exceeded."
+  }
+  conditions {
+    display_name = "(${var.cluster_name}): Signup was throttled"
+    condition_threshold {
+      filter     = "metric.type=\"logging.googleapis.com/user/${google_logging_metric.api_signup_throttling_applied.name}\" AND resource.type=\"k8s_pod\""
+      duration   = "0s"
+      comparison = "COMPARISON_GT"
+      aggregations {
+        alignment_period     = "300s"
```

Comment:
> Probably want to make this similar to the SQL alerting failure 

SQL alerting is where I copied the value from https://github.com/wmde/wbaas-deploy/blob/0a5b6f692d9ad1398ce4682a17f27e9cd3e1eeb1/tf/modules/monitoring/replication-lag-alert.tf#L31. I also think a shorter value might be better as it will bring us closer to telling us how many users are experiencing, instead of telling us "it's broken right now, not sure how many people are affected".

I gotta admit I wonder if those alerts are the right primitive to use for such a thing to begin with. Is there anything in GCP monitoring that is designed for being notified of point-in-time events that cannot be "resolved" in an automatically measurable fashion? I took a brief look, but couldn't find anything.

Reply:
I think what we want is something that works in the way "Sentry" would work, which seems to be close to this https://cloud.google.com/error-reporting/docs/

However, it inverses control and the application would need to push the error to the service. Not sure right now.

Reply:
It looks as if we already have Error Reporting enabled via Stack Driver, we just never use it for anything application level: https://console.cloud.google.com/errors?referrer=search&project=wikibase-cloud

Reply:
Unfortunately it seems terraform doesn't support creating notifications from the Error Reporting service yet https://github.com/hashicorp/terraform-provider-google/issues/12068

Reply:
> SQL alerting is where I copied the value from
Ah, I was unclear. I was thinking about https://github.com/wmde/wbaas-deploy/blob/0a5b6f692d9ad1398ce4682a17f27e9cd3e1eeb1/tf/modules/monitoring/sql-backup-failure-alert.tf#L26 which is another (and functionally different) sql alert.

I think we should try using this error reporting thing; I agree that log based metrics exceeding a count doesn't feel like the right concept for this type of alerting (if it happens even once we would like to know).

Reply:
So how would we do this? Close this PR, update `api` to `report` and see what it brings us?

Reply:
I think that makes sense; let's try to not totally lose these commits though

Reply:
PR in `api` https://github.com/wbstack/api/pull/620